batch/batch/front_end/query/query_v2.py

```diff
@@ -125,40 +125,43 @@ def parse_list_batches_query_v2(user: str, q: str, last_batch_id: Optional[int])
         where_conditions.append(f'({cond})')
         where_args += args
 
-    sql = f"""
-SELECT batches.*,
-  cancelled_t.cancelled IS NOT NULL AS cancelled,
-  job_groups_n_jobs_in_complete_states.n_completed,
-  job_groups_n_jobs_in_complete_states.n_succeeded,
-  job_groups_n_jobs_in_complete_states.n_failed,
-  job_groups_n_jobs_in_complete_states.n_cancelled,
-  cost_t.cost, cost_t.cost_breakdown
-FROM job_groups
-LEFT JOIN batches ON batches.id = job_groups.batch_id
-LEFT JOIN billing_projects ON batches.billing_project = billing_projects.name
-LEFT JOIN job_groups_n_jobs_in_complete_states ON job_groups.batch_id = job_groups_n_jobs_in_complete_states.id AND job_groups.job_group_id = job_groups_n_jobs_in_complete_states.job_group_id
-LEFT JOIN LATERAL (
-  SELECT 1 AS cancelled
-  FROM job_group_self_and_ancestors
-  INNER JOIN job_groups_cancelled
-    ON job_group_self_and_ancestors.batch_id = job_groups_cancelled.id AND
-      job_group_self_and_ancestors.ancestor_id = job_groups_cancelled.job_group_id
-  WHERE job_groups.batch_id = job_group_self_and_ancestors.batch_id AND
-    job_groups.job_group_id = job_group_self_and_ancestors.job_group_id
-) AS cancelled_t ON TRUE
-STRAIGHT_JOIN billing_project_users ON batches.billing_project = billing_project_users.billing_project
-LEFT JOIN LATERAL (
-  SELECT COALESCE(SUM(`usage` * rate), 0) AS cost, JSON_OBJECTAGG(resources.resource, COALESCE(`usage` * rate, 0)) AS cost_breakdown
-  FROM (
+    sql = f"""\
+SELECT batches.*
+     , cancelled_t.cancelled IS NOT NULL AS cancelled
+     , job_groups_n_jobs_in_complete_states.n_completed
+     , job_groups_n_jobs_in_complete_states.n_succeeded
+     , job_groups_n_jobs_in_complete_states.n_failed
+     , job_groups_n_jobs_in_complete_states.n_cancelled
+     , cost_t.cost
+     , cost_t.cost_breakdown
+FROM batches IGNORE INDEX (batches_deleted)
```

Comment:
I have a slight forward-looking concern over introducing so much mysql-specific syntax to the query. I have a suspicion that postgres support might be a useful option to have in certain potential terra futures. Not enough to block this, but enough to make me mildly uneasy...

Reply:
If I remember correctly, rawls and sam use MySQL (or cloud sql). Anyway, all that is kind of irrelevant as
- different databases have different optimisers and may plan queries differently
- many of the queries in batch are very tuned to MySQL (eg `STRAIGHT_JOIN`, `LATERAL`, `(a, b, ..) IN Expression`, etc) 
- we'd therefore have to consider using different DBMSs very carefully.

Reply:
Point well taken that this is not "the thin end of the wedge" and more like "a pattern already throughout batch". Point also well taken that adding support is not necessarily a trivial thing to do. So 👍 for following the pattern here.

For more context on my thoughts here - there are certainly 'ways to terra' that do not go through psql, but our terra-on-azure app, for example, cannot have a managed database because of this reliance, though maybe that's more because it was never added than some fundamental blocker

Reply:
Also I have a natural prior that 'all this optimization for mysql' is just not as necessary in psql because their optimizer is less likely to do the weird things that mysql does. But there's not data behind that, just biases.

Reply:
> our terra-on-azure app, for example, cannot have a managed database because of this reliance, though maybe that's more because it was never added than some fundamental blocker

That's good to know for future work, thanks for sharing.